### DISCUSSION TITLE
Fullframegraphics: easily make full frame graphics for Beamer slideshows

### DISCUSSION THREAD

**Discussion by pagiraud** [quarto-dev/quarto-cli#9983] [Show and Tell]
### Description

Last weekend, I wrote [a filter](https://github.com/pagiraud/fullframegraphics) that makes it easy to produce full frame graphics for Beamer slideshows.

Usage is very simple : 

```markdown
## Title of the full frame slide {.fullframegraphic}

![Toronto Skyline Wide 2014 by Christine Wagner (Flickr)](images/Toronto_Skyline_Wide_2014.jpg)
```

will be rendered:

![PDF slide](https://github.com/pagiraud/fullframegraphics/raw/main/images/example2.webp)

For years, I’ve relied on pure LaTeX Beamer to produce my slideshows. It has always been a perfect fit for me but for one aspect: it is not straightforward to make full frame graphics. A few years ago, I found [a little piece of code](https://www.ojdo.de/wp/2018/06/finally-the-definitive-full-frame-graphic-commands-for-beamer-in-latex/) by Ojdo that really did exactly what I wanted.

But now I write in .md and converts the document to Beamer. Of course, you can directly insert LaTeX code in a markdown document, but I wanted to use pure markdown syntax with a Lua filter to:
- automatically determine whether the image should be adjusted vertically or horizontally
- see an image preview in Wysiwym editors such as RStudio or Zettlr.
- configure the caption style easily.

I know I will use this filter all the time, but I’m not sure if there are still a lot of Beamer output users nowadays. I hope this filter can be useful for others than me.

All kinds of feedbacks are of course welcome!

**Comment by cderv** [comment]
Thanks a lot for sharing ! I'll add to our listing: https://quarto.org/docs/extensions/listing-filters.html
